superset-frontend/src/pages/ExportGoogleSheets/index.tsx

```diff
@@ -0,0 +1,76 @@
+/**
```

Comment:
This blank page appears to be superfluous. It might be more efficient to incorporate this logic within the SQL Lab page and only display the page for the Google Sheets link if the operation is successful.

Reply:
My reasoning is that using an separate page provides a better user experience, and helps decouple the code.

It prevents popup blocker issues, but importantly, this process can take some time, especially as the backend might need to rerun a query for export:
https://github.com/apache/superset/blob/9033e72679fad7efdabc0f5b28607d974a7b5e1f/superset/commands/sql_lab/export.py?plain=1#L112-L126

Our databases (SparkSQL mainly) are particularly slow, and then adding on top of that expensive queries and large amounts of data that need to be moved, we're looking at a 1m average wait time, with the upper bounds being around 15m.

By having this process occur on a separate page it allows the export to happen asynchronously, and for the user to continue their use of Superset while the export occurs.
Notably the CSV export already behaves like this - the download starts asynchronously to the UI.

Simple alternatives would be a modal, or hover popup, suffering from the long loading time issue.

A more complex alternative would be to perform the export on the backend, and provide a notification to the user when it's done. I'm not sure if Superset has this kind of thing set up.

Reply:
> A more complex alternative would be to perform the export on the backend, and provide a notification to the user when it's done. I'm not sure if Superset has this kind of thing set up.

I guess I could do this in the front-end with toasts actually, that would be a viable alternative to me.

But I still prefer the UX and simplicity of the separate page.

Reply:
> I guess I could do this in the front-end with toasts actually, that would be a viable alternative to me.

After experimenting with this I've found that Superset navigation can unintuitively trigger a full page reload, e.g. navigating from "SQL Lab" to "Datasets", meaning that the front-end export process would be lost.

This could be mitigated by a prevent-navigate-away alert, but I don't think that would be preferable UX, and we're heading further away from simplicity.

Reply:
@justinpark Since there has been no feedback on this, I'm going to go with the loading page as it seems to be the best option.